packages/php-wasm/compile/php/php_wasm.c

```diff
@@ -344,6 +344,8 @@ EM_JS(__wasi_errno_t, js_fd_read, (__wasi_fd_t fd, const __wasi_iovec_t *iov, si
             }
 
             const success = returnCode === 0;
+            // @TODO: Do not reason about child_proc_by_fd. Just use pipes and detect
```

Comment:
Actually, PHP uses that distinction internally so maybe it's fine?


https://github.com/php/php-src/blob/004cb827501d1ddaf98daacb185a53e0816f78a7/main/streams/plain_wrapper.c#L476

```c

			if (data->is_process_pipe) {
				errno = 0;
				ret = pclose(data->file);

#if HAVE_SYS_WAIT_H
				if (WIFEXITED(ret)) {
					ret = WEXITSTATUS(ret);
				}
#endif
			} else {
				ret = fclose(data->file);
				data->file = NULL;
			}
```